### DISCUSSION TITLE
Issue with Installing App via Apple Business Manager on User Enrolled Devices

### DISCUSSION THREAD

**Discussion by ashishmeher216** [micromdm/nanomdm#154] [General]
Hello everyone,

I've set up a User Enrollment flow that allows users to sign in with a Managed Apple ID using the "Sign in with Work or School Account" option. This enables them to download a profile and enroll their personal devices.

However, I'm encountering issues when attempting to install an app purchased and licensed through Apple Business Manager (ABM). As far as I understand, ABM / VPP  is the only available method to install apps on devices enrolled via User Enrollment.

Here is the command I'm using to install the application, which exists in my ABM account:
`<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
  <dict>
    <key>Command</key>
    <dict>
      <key>RequestType</key>
      <string>InstallApplication</string>
      <key>iTunesStoreID</key>
      <integer>951937596</integer>
      <key>InstallAsManaged</key>
      <true/>
      <key>ManagementFlags</key>
      <integer>5</integer>
      <key>Options</key>
      <dict>
        <key>PurchaseMethod</key>
        <integer>1</integer>
      </dict>
      <key>Attributes</key>
      <dict>
        <key>VPNUUID</key>
        <string>1234-1234-9876-9876</string>
      </dict>
    </dict>
    <key>CommandUUID</key>
    <string>9775c608-6105-4325-8084-a6d111cf5d70</string>
  </dict>
</plist>
`

When I run this command, I receive the error message: "Could not retrieve license for the app with iTunes Store ID 951937596."

Does anyone know if I need to pass licensing details in the command? I also have a content token from ABM's Apps and Books section, but I'm unsure how to utilize it to facilitate app installation.

Any insights would be greatly appreciated! Thank you.

**Comment by jessepeterson** [comment]
Sounds like you're trying to install VPP apps. Yes, you'll need to use the [Apps and Books (or legacy VPP) APIs](https://developer.apple.com/documentation/devicemanagement/app_and_book_management) to manage licenses for devices or users to install onto device enrollments.
